### PR TITLE
Initial attempt to get JackTrip on FreeBSD

### PR DESCRIPTION
**Pull request by psiborg112**
Change preprocessor definitions in an initial attempt to get JackTrip running on FreeBSD

### CODE REVIEW

**Comment by psiborg112** [comment]
@ntonnaett, clang-format is proposing some changes here in main that don't seem to make sense. It's pulling the #else part of a couple of preprocessor directives back an indent, and I'm not sure why. (This 'correction' was in the code previously, but I thought it was an oversight and manually adjusted things in this upload.)

**Comment by yurivict** [comment]
Somehow the patches don't apply to 1.5.1.

**Comment by psiborg112** [comment]
They won't. You'll need to pull this branch from the repository. (If you have git installed then you can just use the following command line: git clone --branch freebsd https://github.com/jacktrip/jacktrip.git)

**Comment by yurivict** [comment]
The ```freebsd``` branch builds usimg cmake and then the link fails:
```
ld: error: undefined symbol: RtAudio::~RtAudio()
>>> referenced by RtAudioInterface.cpp
>>>               CMakeFiles/jacktrip.dir/src/RtAudioInterface.cpp.o:(RtAudioInterface::~RtAudioInterface())
>>> referenced by RtAudioInterface.cpp
>>>               CMakeFiles/jacktrip.dir/src/RtAudioInterface.cpp.o:(RtAudioInterface::~RtAudioInterface())
>>> referenced by RtAudioInterface.cpp
>>>               CMakeFiles/jacktrip.dir/src/RtAudioInterface.cpp.o:(RtAudioInterface::getDefaultDevice(bool))
>>> referenced 11 more times

ld: error: undefined symbol: RtAudio::RtAudio(RtAudio::Api)
>>> referenced by RtAudioInterface.cpp
>>>               CMakeFiles/jacktrip.dir/src/RtAudioInterface.cpp.o:(RtAudioInterface::setup())
>>> referenced by RtAudioInterface.cpp
>>>               CMakeFiles/jacktrip.dir/src/RtAudioInterface.cpp.o:(RtAudioInterface::getDefaultDevice(bool))
>>> referenced by RtAudioInterface.cpp
>>>               CMakeFiles/jacktrip.dir/src/RtAudioInterface.cpp.o:(RtAudioInterface::printDeviceInfo(unsigned int))
>>> referenced 4 more times

ld: error: undefined symbol: RtAudio::openStream(RtAudio::StreamParameters*, RtAudio::StreamParameters*, unsigned long, unsigned int, unsigned int*, int (*)(void*, void*, unsigned int, double, unsigned int, void*), void*, RtAudio::StreamOptions*, void (*)(RtAudioError::Type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&))
>>> referenced by RtAudioInterface.cpp
>>>               CMakeFiles/jacktrip.dir/src/RtAudioInterface.cpp.o:(RtAudioInterface::setup())
c++: error: linker command failed with exit code 1 (use -v to see invocation)
```

rtaudio-5.1.0 is installed, which provides ```rtaudio.pc```.

**Comment by psiborg112** [comment]
Use the ./build script for a qmake build. Or meson. (I'm pretty sure no one other than me uses cmake to build the project so it's only tested on my development set up.)

The fact that it builds though is very encouraging.

**Comment by yurivict** [comment]
It builds fine with meson.

When you would merge this branch and make a release I would be happy to pick it up into the port.

File: src/main.cpp [hunk 1]
```diff
@@ -97,7 +97,7 @@ QCoreApplication* createApplication(int& argc, char* argv[])
             return new QCoreApplication(argc, argv);
         }
 #else
-#ifdef __linux__
+#if !(defined(_WIN32) || defined(__APPLE__))
```
Comment (ntonnaett) [inline src/main.cpp:100]:
```suggestion
#if defined(__unix__)
```
That's more clear, I think. On https://erato1004.tistory.com/entry/Predefined-Compiler-Macros it states that macOS sets `BSD` but not `__unix__`.

**Comment by ntonnaett** [comment]
The rest looks good to me. A small side quest(ion): Why don't we use SO_REUSEPORT on linux? It's supported since kernel version 3.9.

File: src/main.cpp (hunk 1 above)
Comment (psiborg112) [reply src/main.cpp:100]: Sure. I thought MacOS set __unix__, but seems it doesn't.

Comment (psiborg112) [reply src/main.cpp:100]: Although on the plus side the current option means we are only ever relying on the state of _WIN32, \_\_APPLE\_\_, and \_\_linux\_\_

Comment (ntonnaett) [reply src/main.cpp:100]: That's true. I wanted to change that because I initially misread it.

Comment (ntonnaett) [reply src/main.cpp:100]: Maybe we could change `!` to `not`? Shouldn't that work?

Comment (psiborg112) [reply src/main.cpp:100]: Yeah, totally understand that. (Reordered some later ones for that exact same reason.) If you'd prefer \_\_unix\_\_ then I'm happy to go with that

Comment (psiborg112) [reply src/main.cpp:100]: Yeah, totally understand that. (Reordered some later ones for that exact same reason.) If you'd prefer \_\_unix\_\_ then I'm happy to go with that.

Comment (ntonnaett) [reply src/main.cpp:100]: My main concern is the `!`. So I would be okay with `not` but I would prefer the `__unix__`.

Comment (psiborg112) [reply src/main.cpp:100]: Let's go with \_\_unix\_\_ then.

**Comment by psiborg112** [comment]
Because I guess we don't really want SO_REUSEPORT. We don't need multiple sockets to be bound to the port, we just want to be able to bind to it when it's in TIME_WAIT. I guess the real question is whether we still need SO_REUSEPORT on MacOS?

**Review by ntonnaett — APPROVED** [review]

**Comment by ntonnaett** [comment]
We should open a discussion/issue about the SO_REUSEPORT.

**Pull request merged**